sql/core/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveDefaultColumnsSuite.scala

```diff
@@ -24,7 +24,7 @@ import org.apache.spark.sql.test.SharedSparkSession
 import org.apache.spark.sql.types.{StructField, StructType, TimestampType}
 
 class ResolveDefaultColumnsSuite extends QueryTest with SharedSparkSession {
-  val rule = ResolveDefaultColumns(catalog = null)
+  val rule = ResolveDefaultColumns(null)
```

Comment:
why passing a `null` here and the new test case still passes?  I wonder if the code changes are tested.

Reply:
Good question, as discussed offline: This `rule` field in this unit test just exercises the `addMissingDefaultValuesForInsertFromInlineTable` method directly. This never uses the `resolveRelation` method anyway. This applies to the first two test cases only.  (I prototyped moving `ResolveDefaultColumns` out of the `Analyzer` class to make passing this unnecessary, but this proved more complex than the scope of this PR; I can explore separately).

The last two cases use SQL to cover the new code changes, which exercises the `ResolveDefaultColumns` rule instance in the analyzer instead.